File: app/src/main/java/com/boogie/vibestation/models/Playlist.java
package com.boogie.vibestation.models;

import java.util.ArrayList;
import java.util.Objects;

/**
 * Represents a user-customizable playlist containing a list of Songs.
 */
public class Playlist {
    public String name;
    public String imageUri;
    public String description;
    public boolean isFire = false;
    public final ArrayList<Song> songs = new ArrayList<>();

    /**
     * Constructs a Playlist instance with initial name and optional cover image URI.
     */
    public Playlist(String name, String imageUri) {
        this.name = name;
        this.imageUri = imageUri;
        this.description = "";
    }

    /**
     * Checks equality based on unique playlist name.
     *
     * @param o Comparison object.
     * @return True if objects represent the same playlist by name.
     */
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Playlist playlist = (Playlist) o;
        return Objects.equals(name, playlist.name);
    }

    /**
     * Computes hash code from playlist name.
     *
     * @return Integer hash code.
     */
    @Override
    public int hashCode() {
        return Objects.hash(name);
    }
}

File: app/src/main/java/com/boogie/vibestation/models/Playlist.kt
package com.boogie.vibestation.models

import java.util.Objects

/**
 * Represents a user-customizable playlist containing a list of Songs.
 * Identity is the playlist name alone.
 */
class Playlist(
    @JvmField var name: String,
    @JvmField var imageUri: String?
) {
    @JvmField
    var description: String = ""

    @JvmField
    var isFire: Boolean = false

    @JvmField
    val songs: ArrayList<Song> = ArrayList()

    /**
     * Checks equality based on unique playlist name.
     *
     * @param other Comparison object.
     * @return True if objects represent the same playlist by name.
     */
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        other as Playlist
        return name == other.name
    }

    /** Computes hash code from playlist name. */
    override fun hashCode(): Int = Objects.hash(name)
}
